web/modules/custom/joinup_discussion/src/EventSubscriber/WorkflowTransitionEventSubscriber.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Drupal\joinup_discussion\EventSubscriber;
+
+use Drupal\comment\Plugin\Field\FieldType\CommentItemInterface;
+use Drupal\state_machine\Event\WorkflowTransitionEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Event subscriber to close comments on archived discussions.
+ */
+class WorkflowTransitionEventSubscriber implements EventSubscriberInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents() {
+    return [
+      'discussion.disable.pre_transition' => 'closeComments',
```

Comment:
Post-transition is too late, it will force a second save on the entity. We already handle these things on pre-transitions for things like new revision and published state.